File: mockserver-client-php/src/Times.php
<?php

declare(strict_types=1);

namespace MockServer;

/**
 * Controls how many times an expectation can be matched.
 */
class Times implements \JsonSerializable
{
    private ?int $remainingTimes;
    private bool $unlimited;

    private function __construct(?int $remainingTimes, bool $unlimited)
    {
        $this->remainingTimes = $remainingTimes;
        $this->unlimited = $unlimited;
    }

    /**
     * Match unlimited times.
     */
    public static function unlimited(): self
    {
        return new self(null, true);
    }

    /**
     * Match exactly N times.
     */
    public static function exactly(int $count): self
    {
        return new self($count, false);
    }

    /**
     * Match once.
     */
    public static function once(): self
    {
        return self::exactly(1);
    }

    /**
     * @return array<string, mixed>
     */
    public function jsonSerialize(): array
    {
        return $this->toArray();
    }

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        $data = ['unlimited' => $this->unlimited];
        if ($this->remainingTimes !== null) {
            $data['remainingTimes'] = $this->remainingTimes;
        }
        return $data;
    }
}

File: mockserver-client-dotnet/src/MockServer.Client/Models/Times.cs
using System.Text.Json.Serialization;

namespace MockServer.Client.Models;

/// <summary>
/// Specifies how many times an expectation should be used before it is discarded.
/// </summary>
public sealed class Times
{
    [JsonPropertyName("remainingTimes")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? RemainingTimes { get; set; }

    [JsonPropertyName("unlimited")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public bool? IsUnlimited { get; set; }

    /// <summary>
    /// Creates a Times that matches exactly <paramref name="count"/> times.
    /// </summary>
    public static Times Exactly(int count) => new() { RemainingTimes = count, IsUnlimited = false };

    /// <summary>
    /// Creates a Times that matches unlimited times.
    /// </summary>
    public static Times Unlimited() => new() { IsUnlimited = true };

    /// <summary>
    /// Creates a Times that matches exactly once.
    /// </summary>
    public static Times Once() => Exactly(1);
}
